subsys/net/ip/CMakeLists.txt

```diff
@@ -27,6 +27,7 @@ zephyr_library_sources_ifdef(CONFIG_NET_STATISTICS   net_stats.c)
 zephyr_library_sources_ifdef(CONFIG_NET_TCP          connection.c tcp.c)
 zephyr_library_sources_ifdef(CONFIG_NET_TRICKLE      trickle.c)
 zephyr_library_sources_ifdef(CONFIG_NET_UDP          connection.c udp.c)
+zephyr_library_sources_ifdef(CONFIG_NET_SOCKETS_PACKET  connection.c)
```

Comment:
This might give compiler error if TCP and SOCKETS_PACKET is defined at the same time (I did not test this).